src/main/resources/static/js/testrules.js

```diff
@@ -0,0 +1,346 @@
+          if (!eventsValid) {
+            alert("Events are not a valid json format");
+          } else {
+            var ajaxHttpSender = new AjaxHttpSender();
+            //console.log(JSON.stringify(JSON.parse('{"listRulesJson":' + JSON.stringify(formRules) + ',"listEventsJson":' + events.toString() + '}')));
+            ajaxHttpSender.sendAjax(frontendServiceUrl + "/rules/rule-check/aggregation", "POST", JSON.stringify(JSON.parse('{"listRulesJson":'
+                + JSON.stringify(formRules) + ',"listEventsJson":' + events.toString() + '}')), callback);
+          }
+        };
+
+        // This function for adding rule
+        self.addRule = function(viewModel, event) {
+          var newValue = event.target.value;
+          if (newValue != '') {
+            self.rulesBindingList.push({
+              "TemplateName" : "",
+              "Type" : newValue
+
+            });
+            self.removeDropdown(newValue);
+          }
+        }
+        return self;
+      }
+
+      
+      var vm = new AppViewModel([]);
+      ko.applyBindings(vm, $("#submitButton")[0]);
+      vm.rulesBindingList.push({
+        "TemplateName" : "",
+        "Type" : "EiffelArtifactCreatedEvent"
+
+      });
+      vm.removeDropdown("EiffelArtifactCreatedEvent");
+      ko.applyBindings(vm, $("#testTulesDOMObject")[0]);
+
+      //Upload events list json data
+      $(".container").on("click","button.upload_rules", function(event) {
+            event.stopPropagation();
+            event.preventDefault();
+
+            function validateJsonAndCreateSubscriptions(subscriptionFile) {
+              var reader = new FileReader();
+              reader.onload = function() {
+                var fileContent = reader.result;
+                var jsonLintResult = "";
+                try {
+                  jsonLintResult = jsonlint.parse(fileContent);
+                } catch (e) {
+                  $.alert("JSON Format Check Failed:\n" + e.name + "\n" + e.message);
+                  return false;
+                }
+                $.jGrowl('JSON Format Check Succeeded', {
+                  sticky : false,
+                  theme : 'Notify'
+                });
+
+                var rulesList = JSON.parse(fileContent);
+                ko.cleanNode($("#testTulesDOMObject")[0]);
+                ko.cleanNode($("#submitButton")[0]);
+
+                vm.rulesBindingList.removeAll();
+                $('.eventsListDisplay > div:gt(0)').remove();
+                vm.rulesBindingList = ko.observableArray(rulesList);
+                vm.dropdown([]);
+                vm.dropdown = ko.observableArray([ "EiffelArtifactCreatedEvent", "EiffelArtifactPublishedEvent",
+                    "EiffelConfidenceLevelModifiedEvent", "EiffelTestCaseStartedEvent", "EiffelTestCaseFinishedEvent",
+                    "EiffelCompositionDefinedEvent", "EiffelSourceChangeCreatedEvent", "EiffelSourceChangeSubmittedEvent" ]);
+
```

Comment:
Do ko.cleanNode manage to clean this observableArray?. All values in List might also need to be declared as ko.observable  .

Reply:
Yes ko.cleanNode will clean the nodes, but i am modifying the dropdown list elements.
this commented step for reset the dropdown's all values.
I will explain this in demo.